jax/_src/pallas/pallas_call.py

```diff
@@ -164,142 +163,141 @@ def _get_next_indices(grid, indices):
     next_indices.append(jnp.where(carry, 0, i))
   return tuple(reversed(next_indices))
 
-def _pallas_call_impl(*args, jaxpr, name, out_shapes,
-                      interpret, debug: bool,
-                      in_shapes,
-                      input_output_aliases: tuple[tuple[int, int], ...],
-                      grid_mapping: GridMapping,
-                      compiler_params: Any):
+def _pallas_call_impl(*args, **kwargs):
```

Comment:
Can we avoid calling `def_impl` entirely and get an error from the Primitive itself (since `impl` will be `None`)?

Reply:
I prefer it this way: it adds minimal code but allows me to add a comment next to "assert False" of why we do not need an impl rule.